### DISCUSSION TITLE
netbox topology viewer

### DISCUSSION THREAD

**Discussion by por94405** [netbox-community/netbox#10263] [General]
Hello everyone, I installed  netbox topology viewer plugin, when I try to find something by the filter, it gives the following error. What to do?

<class 'django.core.exceptions.FieldError'>

Cannot resolve keyword '_termination_a_device_id' into field. Choices are: _abs_length, color, created, custom_field_data, id, journal_entries, label, last_updated, length, length_unit, status, tagged_items, tags, tenant, tenant_id, terminations, type

Python version: 3.9.6
NetBox version: 3.3.1

**Comment by kkthxbye-code** [comment]
You need v3.0.0 of netbox-topology-views for netbox version 3.3.0 which is in beta right now. So you need to install 3.0.0b2 or wait until it's out of beta.

https://pypi.org/project/netbox-topology-views/#history

https://github.com/mattieserver/netbox-topology-views/releases

**Reply by por94405** [reply]
I agree, but how to install exactly the third version, what is the installation syntax for centos ?

**Reply by a084ed22** [reply]
add "netbox-topology-views==3.0.0b2" on local_requirements.txt

**Reply by por94405** [reply]
> add "netbox-topology-views==3.0.0b2" on local_requirements.txt

That is, I do a normal installation like on github, and then just change the file local_requirements.txt ???

**Reply by por94405** [reply]
Unfortunately it didn't help

**Reply by a084ed22** [reply]
You need to re-issue the upgrade.sh script every time a new plugin is added and then restart netbox.service and netbox-rq.service.

**Reply by por94405** [reply]
restart did, but what about the script, what kind of script? I'm new to this so I'm sorry but I'm wrong.

**Comment by ekrichbaum** [comment]
Go to your netbox dir (/opt/netbox)

source venv/bin/activate

pip install netbox-topology-views --upgrade –pre

cd netbox

python3 manage.py collectstatic --no-input

 

Collect static will grab the new icons for you.

 

Eric

**Comment by por94405** [comment]
thanks everyone, I solved the problem by updating the version of python

**Comment by Marvo2011** [comment]
Which exact version of python do you have? For me it sucks on Python 3.9.2
